### ISSUE TITLE
branch = 12.0-development not exists

### ISSUE THREAD

**Issue by andy380743909** [love2d/love-android#281]
https://github.com/love2d/love-android/blob/dcd27223bf3e701a537109e99b46f3bcce77e623/.gitmodules#L4

**Comment by MikuAuahDark** [comment]
Any error message when you try to clone the repository? Because as far as GHA aware, there seems to be no issue with it.

See https://github.com/love2d/love-android/actions/runs/11311997163/job/31458886727#step:4:81, that points to https://github.com/love2d/love/commit/72d865eedc42fb4adeb6a9bf2ac793c7556fe108.

**Comment by Endaris** [comment]
If I do a `git submodule update --remote` on the main branch of my local clone I get this result:
```
Schwerwiegend: Konnte refs/remotes/origin/12.0-development Commit in Submodul-Pfad 'app/src/main/cpp/love' nicht finden
```
Which is basically saying it couldn't find the branch commit for the submodule path. I assume people would want to run this if they are trying to compile an apk with the most up-to-date love without you having to advance the commit on the submodule on the remote.
I can confirm that the setup the GHA use execute just fine locally.
I assume the difference is that the commands for initializing and syncing only look for the commit specified by the hash in the upstream remote and to find said commit, git only needs to ask the remote url for the commit hash without specifying the branch.
But when trying to update the submodule from the remote it actually looks for the branch to find the commit at its HEAD and then fails.

**Comment by MikuAuahDark** [comment]
Should be fixed in 18d3e8e7cb6661aa7acba3eced4649f9f6481cbf.
